runners/twister2/src/main/java/org/apache/beam/runners/twister2/Twister2LegacyRunner.java

```diff
@@ -0,0 +1,339 @@
+
+/**
+ * A {@link PipelineRunner} that executes the operations in the pipeline by first translating them
+ * to a Twister2 Plan and then executing them either locally or on a Twister2 cluster, depending on
+ * the configuration.
+ */
+public class Twister2LegacyRunner extends PipelineRunner<PipelineResult> {
+
+  private static final Logger LOG = Logger.getLogger(Twister2LegacyRunner.class.getName());
+  private static final String SIDEINPUTS = "sideInputs";
+  private static final String LEAVES = "leaves";
+  private static final String GRAPH = "graph";
+  /** Provided options. */
+  private final Twister2PipelineOptions options;
+
+  public Twister2LegacyRunner(Twister2PipelineOptions options) {
+    this.options = options;
+  }
+
+  public static Twister2LegacyRunner fromOptions(PipelineOptions options) {
+    Twister2PipelineOptions pipelineOptions =
+        PipelineOptionsValidator.validate(Twister2PipelineOptions.class, options);
+    if (pipelineOptions.getFilesToStage() == null) {
+      pipelineOptions.setFilesToStage(
+          detectClassPathResourcesToStage(
+              Twister2LegacyRunner.class.getClassLoader(), pipelineOptions));
+      LOG.info(
+          "PipelineOptions.filesToStage was not specified. "
+              + "Defaulting to files from the classpath: will stage {} files. "
+              + "Enable logging at DEBUG level to see which files will be staged"
+              + pipelineOptions.getFilesToStage().size());
+    }
+    return new Twister2LegacyRunner(pipelineOptions);
+  }
+
+  @Override
+  public PipelineResult run(Pipeline pipeline) {
+    // create a worker and pass in the pipeline and then do the translation
+    Twister2PipelineExecutionEnvironment env = new Twister2PipelineExecutionEnvironment(options);
+    LOG.info("Translating pipeline to Twister2 program.");
+    pipeline.replaceAll(getDefaultOverrides());
+    env.translate(pipeline);
+    setupSystem(options);
+
+    Config config = ResourceAllocator.loadConfig(new HashMap<>());
+
+    JobConfig jobConfig = new JobConfig();
+    jobConfig.put(SIDEINPUTS, extractNames(env.getSideInputs()));
+    jobConfig.put(LEAVES, extractNames(env.getLeaves()));
+    jobConfig.put(GRAPH, env.getTSetGraph());
+
+    int workers = options.getParallelism();
+    Twister2Job twister2Job =
+        Twister2Job.newBuilder()
+            .setJobName(options.getJobName())
+            .setWorkerClass(BeamBatchWorker.class)
+            .addComputeResource(options.getWorkerCPUs(), options.getRamMegaBytes(), workers)
+            .setConfig(jobConfig)
+            .build();
+    Twister2JobState jobState = Twister2Submitter.submitJob(twister2Job, config);
+
+    Twister2PipelineResult result = new Twister2PipelineResult();
+    if (jobState.getJobstate() == DriverJobState.FAILED
+        || jobState.getJobstate() == DriverJobState.RUNNING) {
+      throw new RuntimeException("Pipeline execution failed", jobState.getCause());
```

Comment:
This is something that needs to be fixed form the Twister2 side first, currently the submit method returns once the job is complete, and if it returns with the state as "RUNNING" it means that the job terminated and did not complete. This can be fixed once the issue is fixed on the Twister2 side. I hope keeping it as is for now is OK for now.

Reply:
Thanks for the clarification!